### PR TITLE
Fix plotting series when the first data point is empty.

### PR DESCRIPTION
**Pull request by tonyhignett**
- Add `graphql` to the dependencies, to help interacting with the schema.
- Get the schema in parallel with the first data query and remember it.
- Use it to find the numeric fields under the data paths, instead of inspecting the first data point.

### CODE REVIEW

**Comment by tonyhignett** [comment]
I haven't tried to write any UTs yet; I wanted someone else to have a look first.

File: src/DataSource.ts [hunk 1]
```diff
@@ -19,28 +19,31 @@ import {
...
-    super(instanceSettings);
-    this.basicAuth = instanceSettings.basicAuth;
-    this.withCredentials = instanceSettings.withCredentials;
-    this.url = instanceSettings.url;
+class _RequestFactory implements RequestFactory {
+  constructor(
+    private basicAuth: string | undefined,
+    private withCredentials: boolean | undefined,
+    private url: string,
+    private backendSrv: any
```
Comment (JonathanWylie) [inline src/DataSource.ts:38]: should have a comma at the end here

File: package.json [hunk 2]
```diff
@@ -18,6 +18,6 @@
     "moment": "*"
   },
   "dependencies": {
-    "@types/lodash": "^4.14.144"
```
Comment (JonathanWylie) [inline package.json:21]: Did you mean to remove lodash?

File: dist/module.js [hunk 3]
```diff
@@ -1,3200 +1,3 @@
-define(["@grafana/data","@grafana/runtime","@grafana/ui","lodash","react"], function(__WEBPACK_EXTERNAL_MODULE__grafana_data__, __WEBPACK_EXTERNAL_MODULE__grafana_runtime__, __WEBPACK_EXTERNAL_MODULE__grafana_ui__, __WEBPACK_EXTERNAL_MODULE_lodash__, __WEBPACK_EXTERNAL_MODULE_react__) { return /******/ (function(modules) { // webpackBootstrap
```
Comment (JonathanWylie) [inline dist/module.js]: Are we meant to commit the built artifacts?

**Review by JonathanWylie — CHANGES_REQUESTED** [review]
Basically it looks fine to me.
It still needs the tests as you said already.
You'll likely get a better review from the author, instead of us who have never seen this before.

File: package.json (hunk 2 above)
Comment (JonathanWylie) [reply package.json:21]: Without lodash yarn build fails.

File: src/DataSource.ts (hunk 1 above)
Comment (JonathanWylie) [reply src/DataSource.ts:38]: maybe not supposed to do that in typescript, I don't know

Comment (tonyhignett) [reply src/DataSource.ts:38]: If you add it, prettier will remove it.

File: package.json (hunk 2 above)
Comment (tonyhignett) [reply package.json:21]: Not sure what's going on there. I'm pretty sure it's not being used directly, but maybe the grafana dependencies use it. And it's not installing `lodash` itself, but only the Typescript type definitions. I've reverted the change.

File: dist/module.js (hunk 3 above)
Comment (tonyhignett) [reply dist/module.js]: Looks like it. They're in git, and they've been committed frequently in the past.

**Pull request merged**